server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -674,7 +678,15 @@ public void setLastAcceptedState(ClusterState clusterState) {
             try {
                 final ClusterMetadataManifest manifest;
                 if (shouldWriteFullClusterState(clusterState)) {
-                    manifest = remoteClusterStateService.writeFullMetadata(clusterState);
+                    if (clusterState.metadata().clusterUUIDCommitted() == true) {
+                        final Optional<ClusterMetadataManifest> latestManifest = remoteClusterStateService.getLatestClusterMetadataManifest(
+                            clusterState.getClusterName().value(),
+                            clusterState.metadata().clusterUUID()
+                        );
+                        logger.error("Latest manifest is not present in remote store for cluster UUID: {}", clusterState.metadata().clusterUUID());
+                        previousClusterUUID = latestManifest.isPresent() ? latestManifest.get().getPreviousClusterUUID(): ClusterState.UNKNOWN_UUID;
```

Comment:
previousClusterUUID will be stale if this was not a active master but got elected later since previousclusterUUID gets set at the time of node bootstrap.